test/test_c10d.py

```diff
@@ -3171,9 +3171,12 @@ def _test_nccl_errors_blocking(self, func):
         process_group.allreduce(torch.rand(10).cuda(self.rank))
         if self.rank == 0:
             work = process_group.allreduce(torch.rand(10).cuda(self.rank))
-            with self.assertRaises(RuntimeError):
+            try:
                 # Operation would time out in blocking mode.
                 work.wait()
+            except Exception as e:
+                self.assertEqual("Operation timed out!", str(e))
+                a = torch.rand(10).cuda(self.rank)
```

Comment:
What happens if we try another allreduce operation here?

Reply:
I think it will time out unless we want to init process group again with rank 1 and both of rank 0 and rank 1 ran all reduce again

Reply:
let me try that

Reply:
I tried a few things, run another allreduce will just stuck, I think the two processes are not in the clean sync state to recover in our unit tests. Did not successfully to make these two processes in a clean sync state to rerun all reduce so far (PET is doing that)